In [2]:
import numpy as np
import pandas as pd
import pickle as pkl
import matplotlib as plt
import os

In [66]:
with open('pickles\\Removed_Others_Dialogue.pkl', 'rb') as fl:
    df = pkl.load(fl)
df

date                                              title  \
0    May 26, 2020  Excerpts from Speech of President Rodrigo Roa ...   
1    May 25, 2020  Talk to the People of President Rodrigo Roa Du...   
2    May 22, 2020  Speech of President Rodrigo Roa Duterte during...   
3    May 19, 2020  Talk to the People of President Rodrigo Roa Du...   
4    May 12, 2020  Talk to the People of President Rodrigo Roa Du...   
..            ...                                                ...   
712  May 09, 2018  Speech of President Rodrigo Roa Duterte during...   
713  May 08, 2018  Speech of President Rodrigo Roa Duterte during...   
714  May 05, 2018  Speech of President Rodrigo Roa Duterte during...   
715  May 04, 2018  Speech of President Rodrigo Roa Duterte during...   
716  May 02, 2018  Speech of President Rodrigo Roa Duterte during...   

                                                   url  \
0    https://pcoo.gov.ph/presidential-speech/excerp...   
1    https://pcoo.gov.ph/presidential-speech/talk-t...   
2    https://pcoo.gov.ph/presidential-speech/speech...   
3    https://pcoo.gov.ph/presidential-speech/talk-t...   
4    https://pcoo.gov.ph/presidential-speech/talk-t...   
..                                                 ...   
712  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
713  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
714  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
715  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
716  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   

                                                 event  \
0    Meeting with Philippine Army (PA) and Philippi...   
1               On Coronavirus Disease 2019 (COVID-19)   
2    Commencement Exercsies of the Philippine Milit...   
3               On Coronavirus Disease 2019 (COVID-19)   
4               On Coronavirus Disease 2019 (COVID-19)   
..                                                 ...   
712                                                NaN   
713                                                NaN   
714                                                NaN   
715                                                NaN   
716                                                NaN   

                                              location  \
0           Malago Clubhouse, Malacañang Park, Manila    
1                       Malago Clubhouse in Malacañang   
2           Malago Clubhouse, Malacañang Park, Manila    
3                       Malago Clubhouse in Malacañang   
4                       Malago Clubhouse in Malacañang   
..                                                 ...   
712                                   Malacañan Palace   
713                   the Rizal Hall, Malacañan Palace   
714  the Isla Ballroom, EDSA Shangri-La, Mandaluyon...   
715              the SMX Convention Center, Davao City   
716        Liwasang Alfaro G. Aguirre, Mulanay, Quezon   

                                            transcript  
0         So ako pati si Bong during my mayorship d...  
1    PRESIDENT RODRIGO ROA DUTERTE: I remember dist...  
2    Kindly sit down. [May upuan sila? Okay.], Defe...  
3    PRESIDENT RODRIGO ROA DUTERTE: Good evening my...  
4    PRESIDENT DUTERTE: Sir, one question. Itong op...  
..                                                 ...  
712  Kindly sit down. Salamat for the courtesy. Fin...  
713  Maybe after my prepared speech, this is just a...  
714  Thank you. Kindly sit down.     I’d forego wit...  
715  Salamat po. Thank you for your courtesy. You m...  
716  Kindly sit down. Thank you for your courtesy. ...  

[705 rows x 6 columns]

In [67]:
df.reset_index(inplace=True, drop=True)
df

date                                              title  \
0    May 26, 2020  Excerpts from Speech of President Rodrigo Roa ...   
1    May 25, 2020  Talk to the People of President Rodrigo Roa Du...   
2    May 22, 2020  Speech of President Rodrigo Roa Duterte during...   
3    May 19, 2020  Talk to the People of President Rodrigo Roa Du...   
4    May 12, 2020  Talk to the People of President Rodrigo Roa Du...   
..            ...                                                ...   
700  May 09, 2018  Speech of President Rodrigo Roa Duterte during...   
701  May 08, 2018  Speech of President Rodrigo Roa Duterte during...   
702  May 05, 2018  Speech of President Rodrigo Roa Duterte during...   
703  May 04, 2018  Speech of President Rodrigo Roa Duterte during...   
704  May 02, 2018  Speech of President Rodrigo Roa Duterte during...   

                                                   url  \
0    https://pcoo.gov.ph/presidential-speech/excerp...   
1    https://pcoo.gov.ph/presidential-speech/talk-t...   
2    https://pcoo.gov.ph/presidential-speech/speech...   
3    https://pcoo.gov.ph/presidential-speech/talk-t...   
4    https://pcoo.gov.ph/presidential-speech/talk-t...   
..                                                 ...   
700  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
701  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
702  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
703  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
704  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   

                                                 event  \
0    Meeting with Philippine Army (PA) and Philippi...   
1               On Coronavirus Disease 2019 (COVID-19)   
2    Commencement Exercsies of the Philippine Milit...   
3               On Coronavirus Disease 2019 (COVID-19)   
4               On Coronavirus Disease 2019 (COVID-19)   
..                                                 ...   
700                                                NaN   
701                                                NaN   
702                                                NaN   
703                                                NaN   
704                                                NaN   

                                              location  \
0           Malago Clubhouse, Malacañang Park, Manila    
1                       Malago Clubhouse in Malacañang   
2           Malago Clubhouse, Malacañang Park, Manila    
3                       Malago Clubhouse in Malacañang   
4                       Malago Clubhouse in Malacañang   
..                                                 ...   
700                                   Malacañan Palace   
701                   the Rizal Hall, Malacañan Palace   
702  the Isla Ballroom, EDSA Shangri-La, Mandaluyon...   
703              the SMX Convention Center, Davao City   
704        Liwasang Alfaro G. Aguirre, Mulanay, Quezon   

                                            transcript  
0         So ako pati si Bong during my mayorship d...  
1    PRESIDENT RODRIGO ROA DUTERTE: I remember dist...  
2    Kindly sit down. [May upuan sila? Okay.], Defe...  
3    PRESIDENT RODRIGO ROA DUTERTE: Good evening my...  
4    PRESIDENT DUTERTE: Sir, one question. Itong op...  
..                                                 ...  
700  Kindly sit down. Salamat for the courtesy. Fin...  
701  Maybe after my prepared speech, this is just a...  
702  Thank you. Kindly sit down.     I’d forego wit...  
703  Salamat po. Thank you for your courtesy. You m...  
704  Kindly sit down. Thank you for your courtesy. ...  

[705 rows x 6 columns]

In [68]:
df.transcript.nunique()

705

In [69]:
import re

def clean_transcript(single_str):
    lt_names= ['PRESIDENT RODRIGO DUTERTE:', 'PRESIDENT DUTERTE:', 'PRESIDENT RODRIGO ROA DUTERTE:', 'DUTERTE:', 
               'PRES. DUTERTE:', 'PRRD']
    str_cleaned = single_str
    for nm in lt_names:
        str_cleaned = str_cleaned.replace(nm, ' ')
    str_cleaned = str_cleaned.replace('\xa0', ' ')
    str_cleaned = str_cleaned.replace('\x0c', ' ')
    str_cleaned = str_cleaned.replace('\\', ' ')
    str_cleaned = str_cleaned.replace('—', ' ')
    str_cleaned = str_cleaned.replace('-', ' ')
    str_cleaned = str_cleaned.replace('...', ' ')
    str_cleaned = str_cleaned.replace('…', ' ')
    str_cleaned = str_cleaned.replace('END', ' ')
    str_cleaned = re.sub(r"\[(.*?)\]", ' ', str_cleaned) #remove all that is contained within brackets
    str_cleaned = re.sub(r"\((.*?)\)", ' ', str_cleaned) #remove all that is contained within parentheses
    str_cleaned = re.sub(r'\“(.+?)\”', ' ', str_cleaned) #remove all that is in quatation as he probably quoting someone
    str_cleaned = str_cleaned.replace('‘', ' ')
    str_cleaned = [wrd.lower() for wrd in str_cleaned.split(' ') if wrd != '']
    str_cleaned = ' '.join(str_cleaned)

    return str_cleaned

>We're going to apply the function above to every transcript for cleaning. But before that let's correct the indexing

In [70]:
df['transcript'] = df.transcript.map(clean_transcript)
df

date                                              title  \
0    May 26, 2020  Excerpts from Speech of President Rodrigo Roa ...   
1    May 25, 2020  Talk to the People of President Rodrigo Roa Du...   
2    May 22, 2020  Speech of President Rodrigo Roa Duterte during...   
3    May 19, 2020  Talk to the People of President Rodrigo Roa Du...   
4    May 12, 2020  Talk to the People of President Rodrigo Roa Du...   
..            ...                                                ...   
700  May 09, 2018  Speech of President Rodrigo Roa Duterte during...   
701  May 08, 2018  Speech of President Rodrigo Roa Duterte during...   
702  May 05, 2018  Speech of President Rodrigo Roa Duterte during...   
703  May 04, 2018  Speech of President Rodrigo Roa Duterte during...   
704  May 02, 2018  Speech of President Rodrigo Roa Duterte during...   

                                                   url  \
0    https://pcoo.gov.ph/presidential-speech/excerp...   
1    https://pcoo.gov.ph/presidential-speech/talk-t...   
2    https://pcoo.gov.ph/presidential-speech/speech...   
3    https://pcoo.gov.ph/presidential-speech/talk-t...   
4    https://pcoo.gov.ph/presidential-speech/talk-t...   
..                                                 ...   
700  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
701  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
702  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
703  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
704  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   

                                                 event  \
0    Meeting with Philippine Army (PA) and Philippi...   
1               On Coronavirus Disease 2019 (COVID-19)   
2    Commencement Exercsies of the Philippine Milit...   
3               On Coronavirus Disease 2019 (COVID-19)   
4               On Coronavirus Disease 2019 (COVID-19)   
..                                                 ...   
700                                                NaN   
701                                                NaN   
702                                                NaN   
703                                                NaN   
704                                                NaN   

                                              location  \
0           Malago Clubhouse, Malacañang Park, Manila    
1                       Malago Clubhouse in Malacañang   
2           Malago Clubhouse, Malacañang Park, Manila    
3                       Malago Clubhouse in Malacañang   
4                       Malago Clubhouse in Malacañang   
..                                                 ...   
700                                   Malacañan Palace   
701                   the Rizal Hall, Malacañan Palace   
702  the Isla Ballroom, EDSA Shangri-La, Mandaluyon...   
703              the SMX Convention Center, Davao City   
704        Liwasang Alfaro G. Aguirre, Mulanay, Quezon   

                                            transcript  
0    so ako pati si bong during my mayorship days s...  
1    i remember distinctly very well that noong nap...  
2    kindly sit down. , defense secretary delfin lo...  
3    good evening my countrymen. today is may 19, t...  
4    sir, one question. itong opening, limited move...  
..                                                 ...  
700  kindly sit down. salamat for the courtesy. fin...  
701  maybe after my prepared speech, this is just a...  
702  thank you. kindly sit down. i’d forego with th...  
703  salamat po. thank you for your courtesy. you m...  
704  kindly sit down. thank you for your courtesy. ...  

[705 rows x 6 columns]

>Let's parse date features/variables/columns into date object

In [71]:
df['date'] = df.date.map(lambda x: x if x.find('_') == -1 else x[:-2]).tolist()
df['date'] = pd.to_datetime(df['date'])
df

date                                              title  \
0   2020-05-26  Excerpts from Speech of President Rodrigo Roa ...   
1   2020-05-25  Talk to the People of President Rodrigo Roa Du...   
2   2020-05-22  Speech of President Rodrigo Roa Duterte during...   
3   2020-05-19  Talk to the People of President Rodrigo Roa Du...   
4   2020-05-12  Talk to the People of President Rodrigo Roa Du...   
..         ...                                                ...   
700 2018-05-09  Speech of President Rodrigo Roa Duterte during...   
701 2018-05-08  Speech of President Rodrigo Roa Duterte during...   
702 2018-05-05  Speech of President Rodrigo Roa Duterte during...   
703 2018-05-04  Speech of President Rodrigo Roa Duterte during...   
704 2018-05-02  Speech of President Rodrigo Roa Duterte during...   

                                                   url  \
0    https://pcoo.gov.ph/presidential-speech/excerp...   
1    https://pcoo.gov.ph/presidential-speech/talk-t...   
2    https://pcoo.gov.ph/presidential-speech/speech...   
3    https://pcoo.gov.ph/presidential-speech/talk-t...   
4    https://pcoo.gov.ph/presidential-speech/talk-t...   
..                                                 ...   
700  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
701  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
702  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
703  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   
704  http://pcoo.gov.ph/wp-content/uploads/2018/05/...   

                                                 event  \
0    Meeting with Philippine Army (PA) and Philippi...   
1               On Coronavirus Disease 2019 (COVID-19)   
2    Commencement Exercsies of the Philippine Milit...   
3               On Coronavirus Disease 2019 (COVID-19)   
4               On Coronavirus Disease 2019 (COVID-19)   
..                                                 ...   
700                                                NaN   
701                                                NaN   
702                                                NaN   
703                                                NaN   
704                                                NaN   

                                              location  \
0           Malago Clubhouse, Malacañang Park, Manila    
1                       Malago Clubhouse in Malacañang   
2           Malago Clubhouse, Malacañang Park, Manila    
3                       Malago Clubhouse in Malacañang   
4                       Malago Clubhouse in Malacañang   
..                                                 ...   
700                                   Malacañan Palace   
701                   the Rizal Hall, Malacañan Palace   
702  the Isla Ballroom, EDSA Shangri-La, Mandaluyon...   
703              the SMX Convention Center, Davao City   
704        Liwasang Alfaro G. Aguirre, Mulanay, Quezon   

                                            transcript  
0    so ako pati si bong during my mayorship days s...  
1    i remember distinctly very well that noong nap...  
2    kindly sit down. , defense secretary delfin lo...  
3    good evening my countrymen. today is may 19, t...  
4    sir, one question. itong opening, limited move...  
..                                                 ...  
700  kindly sit down. salamat for the courtesy. fin...  
701  maybe after my prepared speech, this is just a...  
702  thank you. kindly sit down. i’d forego with th...  
703  salamat po. thank you for your courtesy. you m...  
704  kindly sit down. thank you for your courtesy. ...  

[705 rows x 6 columns]

>Let's group these records by transcripts(since there are duplciates)

In [117]:
grouped_transcripts = df.groupby('transcript')

>Let's find out whose group contains multiple record-basically a duplicate.

In [118]:
multiples = np.array([])

for i in grouped_transcripts.groups:
    #if it contain more than 1 record then it has duplicate
    if len(grouped_transcripts.get_group(i)) > 1:
        #append the group to multiples
        multiples = np.append(multiples, [i])
len(multiples)

8

>So there are 8 transcripts that have duplicates. I checked them individually and they really are duplicates so I'm going to remove the duplicates.

In [82]:
df_temp = df.copy()

In [119]:
for group in multiples:
    df.drop(index=grouped_transcripts.get_group(group).index.values[1:], inplace=True)
    grouped_transcripts = df.groupby('transcript')

In [120]:
df.reset_index(inplace=True, drop=True)
print('Number of unique transcripts:', df.transcript.nunique())
print('Number of transcripts in our data:', df.shape[0])

Number of unique transcripts: 697
Number of transcripts in our data: 697


>So we have a total of 697 unique transcripts. Let's save this as 'Cleaned.pkl'.

In [124]:
with open('pickles\\Cleaned.pkl', 'wb') as fl:
    pkl.dump(df, fl)